packages/quantification/src/quantification.ts

```diff
@@ -548,6 +471,36 @@ const getGrandfatheredTonneQuantities = ({
   };
 };
 
+/**
+ * Takes the year-over-year difference in soil carbon content across multiple polygons,
+ * sums the future and baseline values individually, then provides the average year-over-year
+ * difference over the first 10 years.
+ *
+ * @returns The average year-over-year change for the first 10 years after the switch year
+ */
+const getTenYearProjectedTonnesPerYear = ({
+  somscAnnualDifferencesBetweenFutureAndBaselineScenariosPerPolygon,
+}: {
+  somscAnnualDifferencesBetweenFutureAndBaselineScenariosPerPolygon: AnnualTotals[];
+}): number => {
+  return divide(
+    somscAnnualDifferencesBetweenFutureAndBaselineScenariosPerPolygon
+      .map((annualDifferencesPerPolygon) => {
+        const years = Object.keys(annualDifferencesPerPolygon)
+          .sort()
+          .map((year) => annualDifferencesPerPolygon[year])
+          .slice(0, 10)
+          .filter((yearValue) => !Number.isNaN(yearValue));
+
+        return years.reduce((prevValue, curValue) => {
+          return add(prevValue, curValue);
+        }, 0);
+      })
+      .reduce((prevValue, curValue) => add(prevValue, curValue), 0),
+    10
```

Comment:
Are there always 10 years? We filter out NaN years for example, so we could be averaging < 10 things, but idk if those just count as effectively zero. Either way, may be more resilient to divide by the length, and replace NaN with zero if that's the case

Reply:
We're validating at the start of `getQuantificationSummary` that there should be 10 values here, but I'll add an error here just in case. We don't want to silently fail as that would leave to bad NRT awards.